lib/ansible/module_utils/basic.py

```diff
@@ -1483,31 +1483,28 @@ def _check_type_dict(self, value):
         raise TypeError('%s cannot be converted to a dict' % type(value))
 
     def _check_type_bool(self, value):
-        if isinstance(value, bool):
-            return value
-
-        if isinstance(value, string_types) or isinstance(value, int):
-            return self.boolean(value)
-
-        raise TypeError('%s cannot be converted to a bool' % type(value))
+        if not isinstance(value, bool):
+            try:
+                return self.boolean(value)
+            except:
+                raise TypeError('%s cannot be converted to a bool' % type(value))
+        return value
 
     def _check_type_int(self, value):
-        if isinstance(value, int):
-            return value
-
-        if isinstance(value, string_types):
-            return int(value)
-
-        raise TypeError('%s cannot be converted to an int' % type(value))
+        if not isinstance(value, int):
+            try:
+                return int(value)
```

Comment:
Also, things like int(True) == 1 and int(False) == 0, in python3 int(b'characters'[0]) == 99....